src/test/java/com/example/crudapi/integrationtest/LocationRestApiIntegrationTest.java

```diff
@@ -0,0 +1,468 @@
+package com.example.crudapi.integrationtest;
+
+import com.github.database.rider.core.api.dataset.DataSet;
+import com.github.database.rider.core.api.dataset.ExpectedDataSet;
+import com.github.database.rider.spring.api.DBRider;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.nio.charset.StandardCharsets;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@DBRider
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class LocationRestApiIntegrationTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Nested
+    class FindByCorner {
+        @Test
+        @DataSet(value = "datasets/locations.yml")
+        @Transactional
+        void 指定したcornerのlocationが取得できステータスコード200を返すこと() throws Exception {
+            String response =
+                    mockMvc.perform(MockMvcRequestBuilders.get("/locations/food"))
+                            .andExpect(MockMvcResultMatchers.status().isOk())
+                            .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+
+            JSONAssert.assertEquals("""
+                    {
+                       "corner": "food",
+                       "location_name": "A",
+                       "place": "left-back",
+                       "creator": "yamada",
+                       "date_created": "2023/08/01"
+                     }
+                        """, response, JSONCompareMode.STRICT
+            );
+        }
+
+        @Test
+        @DataSet(value = "datasets/locations.yml")
+        @Transactional
+        void 指定したcornerが存在しないときに例外をスローしてステータスコード404とエラーメッセージを返すこと() throws Exception {
```

Comment:
例外という言葉はあくまでServiceレイヤーで見た話なので結合テストでは不要かなと
```suggestion
        void 指定したcornerが存在しないときにステータスコード404とエラーメッセージを返すこと() throws Exception {
```